definitions/checks/restore/validate_postgresql_dump_permissions.rb

```diff
@@ -0,0 +1,30 @@
+module Checks::Restore
+  class ValidatePostgresqlDumpPermissions < ForemanMaintain::Check
+    metadata do
+      description 'Validate permissions for PostgreSQL dumps'
+
+      param :backup_dir,
+        'Path to backup directory',
+        :required => true
+      manual_detection
+    end
+
+    def run
+      backup = ForemanMaintain::Utils::Backup.new(@backup_dir)
+      if feature(:instance).postgresql_local? && backup.online_backup?
+        errors = []
+        [:candlepin_dump, :foreman_dump, :pulpcore_dump].each do |dump|
+          if backup.file_map[dump][:present]
+            success = system("runuser - postgres -c 'test -r #{backup.file_map[dump][:path]}'")
+            errors << backup.file_map[dump][:path] unless success
+          end
```

Comment:
Just another way to write parts of this (not required, food for thought):

```suggestion
    next unless backup.file_map[dump][:present]
    unless system("runuser - postgres -c 'test -r #{backup.file_map[dump][:path]}'")
      errors << backup.file_map[dump][:path]
    end
```

Reply:
Like!

Reply:
(besides the bad syntax and all of that, fixed)